File: Node/TypeORM_ORM/src/bll/services/regionService.ts
import { Region } from 'dal/entities/Region';
import { RegionRepository } from '../../dal/repositories/regionRepository';

export class RegionService {
  private regionRepository = new RegionRepository();
  async getAll() {
    return await this.regionRepository.getAll();
  }

  async getById(id: number) {
    return await this.regionRepository.getById(id);
  }

  async create(data: Partial<Region>) {
    return await this.regionRepository.create(data);
  }

  async update(id: number, data: Partial<Region>) {
    return await this.regionRepository.update(id, data);
  }

  async delete(id: number) {
    return await this.regionRepository.delete(id);
  }
}

File: Go/GORM_ORM/bll/services/regionService.go
package services

import (
	"GORM_ORM/dal/entities"
	"GORM_ORM/dal/repositories"
)

type RegionService interface {
	GetAll() ([]entities.Region, error)
	GetByID(id int) (*entities.Region, error)
	Create(region *entities.Region) error
	Update(region *entities.Region) error
	Delete(id int) error
}

type regionService struct {
	repo repositories.RegionRepository
}

func NewRegionService(repo repositories.RegionRepository) RegionService {
	return &regionService{repo: repo}
}

func (s *regionService) GetAll() ([]entities.Region, error) {
	return s.repo.GetAll()
}

func (s *regionService) GetByID(id int) (*entities.Region, error) {
	return s.repo.GetByID(id)
}

func (s *regionService) Create(region *entities.Region) error {
	return s.repo.Create(region)
}

func (s *regionService) Update(region *entities.Region) error {
	return s.repo.Update(region)
}

func (s *regionService) Delete(id int) error {
	return s.repo.Delete(id)
}
